apps/prairielearn/src/sync/course-db.ts

```diff
@@ -405,27 +405,27 @@ export async function loadFullCourse(
   const courseInstanceInfos = await loadCourseInstances(courseDir);
   const courseInstances: Record<string, CourseInstanceData> = {};
   for (const [courseInstanceId, courseInstance] of Object.entries(courseInstanceInfos)) {
-    const courseInstanceAccessibleNowOrInFuture = (courseInstance.data?.allowAccess ?? []).some(
-      (rule) => {
-        const startDate = rule.startDate ? parseAllowAccessDate(rule.startDate) : null;
+    // Check if the course instance is "expired". A course instance is considered
+    // expired if it either has zero `allowAccess` rules (in which case it is never
+    // accessible), or if it has one or more `allowAccess` rules and they all have
+    // an `endDate` that is in the past.
+    const allowAccessRules = courseInstance.data?.allowAccess ?? [];
+    const courseInstanceExpired =
+      allowAccessRules.length === 0 ||
```

Comment:
This is not necessary, `.every()` [returns true if the array is empty](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every#description).

Reply:
Thanks, this is left over from a previous version that used inverted logic.

Reply:
Removed in 865eecd.